Okay this is fun... running `./gradlew merlin-server:test` the first time (i.e. deleting `./.gradle` from the repo) results in the following error
```
> Task :merlin-server:test

MerlinBindingsTest > initializationError FAILED
    java.lang.Error at MerlinBindingsTest.java:39
        Caused by: java.io.IOException at MerlinBindingsTest.java:37
            Caused by: java.io.IOException at MerlinBindingsTest.java:37

ConstraintsDSLCompilationServiceTests > initializationError FAILED
    java.io.IOException at ConstraintsDSLCompilationServiceTests.java:71
        Caused by: java.io.IOException at ConstraintsDSLCompilationServiceTests.java:71
```

But then running `./gradlew merlin-server:test` a second time results in passed tests. Seems like gradle isn't able to find its own downloaded node files during the first run, but finds them during the filesystem traversal in the second run.

--- a/merlin-server/build.gradle
+++ b/merlin-server/build.gradle
@@ -49,19 +49,14 @@ assemble {
 }
 
 test {
-  dependsOn(assembleConstraintsDSLCompiler)
+  dependsOn nodeSetup
+  dependsOn assembleConstraintsDSLCompiler
   useJUnitPlatform {
     includeEngines 'jqwik', 'junit-jupiter'
   }
 
   // Add node bin directory to PATH, helps CI/CD services without node installed
-  projectDir.toPath().resolve('.gradle/nodejs').toFile().listFiles().each {
-    environment 'NODE_PATH', "$it/bin/node"
-  }
-
-  if (System.getenv("NODE_PATH") == null) {
-    environment 'NODE_PATH', "node"
-  }
+  environment 'NODE_PATH', nodeSetup.nodeDir.get().toString().concat("/bin/node")
 
   environment "CONSTRAINTS_DSL_COMPILER_ROOT", projectDir.toPath().resolve('constraints-dsl-compiler')
   environment "CONSTRAINTS_DSL_COMPILER_COMMAND", './build/main.js'

--- a/scheduler-worker/build.gradle
+++ b/scheduler-worker/build.gradle
@@ -70,7 +70,8 @@ assemble {
 }
 
 test {
-  dependsOn(assembleSchedulingDSLCompiler)
+  dependsOn nodeSetup
+  dependsOn assembleSchedulingDSLCompiler
   useJUnitPlatform()
 
   testLogging {
@@ -78,13 +79,7 @@ test {
   }
 
   // Add node bin directory to PATH, helps CI/CD services without node installed
-  projectDir.toPath().resolve('.gradle/nodejs').toFile().listFiles().each {
-    environment 'NODE_PATH', "$it/bin/node"
-  }
-
-  if (System.getenv("NODE_PATH") == null) {
-    environment 'NODE_PATH', "node"
-  }
+  environment 'NODE_PATH', nodeSetup.nodeDir.get().toString().concat("/bin/node")
 
   environment "SCHEDULING_DSL_COMPILER_ROOT", projectDir.toPath().resolve('scheduling-dsl-compiler')
   environment "SCHEDULING_DSL_COMPILER_COMMAND", './build/main.js'

